scripts/lintChanged.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# Lints .ts and .tsx files that have changed in this branch
+
+set -eu
+
+TOP="$(realpath "$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)/..")"
+readonly TOP
+source "${TOP}/scripts/shellUtils.sh"
+
+# Ensure this script is running inside an npm run context
+if [[ -z "${npm_lifecycle_event:-}" ]]; then
+    info "Re-executing this script from an npm context"
+    exec npx -c "NODE_OPTIONS=--max_old_space_size=8192 ${BASH_SOURCE[0]} $*"
+fi
+
+# Fetch the commit history to include the merge-base commit
+git fetch origin main --no-tags
+
+MERGE_BASE_SHA_HASH="$(git merge-base origin/main HEAD)"
+# Check if output is empty or malformed
+if [[ -z "$MERGE_BASE_SHA_HASH" ]] || ! [[ "$MERGE_BASE_SHA_HASH" =~ ^[a-fA-F0-9]{40}$ ]]; then
+    error "git merge-base returned unexpected output"
```

Comment:
NAB probably helpful to show what that output is

```suggestion
    error "git merge-base returned unexpected output: $MERGE_BASE_SHA_HASH"
```